lib/diff-list-view.coffee

```diff
@@ -22,17 +22,34 @@ class DiffListView extends SelectListView
     @panel.show()
     @focusFilterEditor()
 
-  viewForItem: ({oldStart, newStart, oldLines, newLines, lineText}) ->
+  viewForItem: ({oldStart, newStart, oldLines, newLines, lineText, exceedsLimit, richLines}) ->
     $$ ->
       @li class: 'two-lines', =>
-        @div lineText, class: 'primary-line'
+        @div class: 'primary-line', =>
+          for richDiff in richLines
+            if richDiff.newLineNumber is -1
+              # removed line, add '-' and mark as deleted
+              @code "- #{richDiff.line}", class: 'diff-line removed'
+            else if richDiff.oldLineNumber is -1
+              # added line, prepend '+' and mark as new
+              @code "+ #{richDiff.line}", class: 'diff-line added'
+          if exceedsLimit
+            @div "... more lines", class: 'diff-line more-lines'
         @div "-#{oldStart},#{oldLines} +#{newStart},#{newLines}", class: 'secondary-line'
 
   populate: ->
-    diffs = repositoryForPath(@editor.getPath())?.getLineDiffs(@editor.getPath(), @editor.getText()) ? []
+    repo = repositoryForPath(@editor.getPath())
+    diffs = repo?.getLineDiffs(@editor.getPath(), @editor.getText()) ? []
+    richDiffs = getRichDiffsForPath(repo, @editor.getPath(), @editor.getText()) ? []
+    limit = 12 # limit displayed lines per diff, will show message with "... more lines"
     for diff in diffs
-      bufferRow = if diff.newStart > 0 then diff.newStart - 1 else diff.newStart
-      diff.lineText = @editor.lineTextForBufferRow(bufferRow)?.trim() ? ''
+      diff.richLines = [] # corresponding lines for this range diff
+      diff.exceedsLimit = false # whether or not all lines are included
+      for richDiff in richDiffs
```

Comment:
Will update to do it once, since rich diffs are sorted in the same order as diffs.